compiler+runtime/test/cpp/jank/runtime/obj/ratio.cpp

```diff
@@ -0,0 +1,401 @@
+        CHECK(result->data.numerator == 3);
+        CHECK(result->data.denominator == 8);
+        CHECK(result2->data.numerator == 8);
+        CHECK(result2->data.denominator == 3);
+      }
+
+      SUBCASE("Division with Native Real")
+      {
+        auto native_real{ 0.5L };
+        auto result = ratio / native_real;
+        auto result2 = native_real / ratio;
+        CHECK(result == doctest::Approx(1.5));
+        CHECK(result2 == doctest::Approx(1 / 1.5));
+      }
+
+      SUBCASE("Comparison with Integer Pointer")
+      {
+        auto int_ptr{ make_box(1LL) };
+        CHECK(ratio < int_ptr);
+        CHECK(ratio != int_ptr);
+        CHECK(int_ptr > ratio);
+        CHECK(int_ptr != ratio);
+      }
+
+      SUBCASE("Comparison with Native Integer")
+      {
+        auto native_int{ 1LL };
+        CHECK(ratio < native_int);
+        CHECK(ratio != native_int);
+        CHECK(native_int > ratio);
+        CHECK(native_int != ratio);
+      }
+
+      SUBCASE("Comparison with Native Real")
+      {
+        auto native_real{ 0.75L };
+        CHECK(ratio == native_real);
+        CHECK(native_real == ratio);
+      }
+    }
+
+    TEST_CASE("Ratio Mixed Arithmetic and Comparisons")
+    {
+      obj::ratio_data ratio{ 5, 8 };
+      auto native_int{ 3LL };
+      auto native_real{ 0.25L };
+
+      SUBCASE("Complex Arithmetic Chain")
+      {
+        auto result = (ratio + native_int)->data - native_real;
+        CHECK(result == doctest::Approx(3.375));
+      }
+
+      SUBCASE("Mixed Comparison")
+      {
+        auto real_ptr{ make_box(1.0L) };
+        auto int_ptr{ make_box(1LL) };
+
+        CHECK(ratio < real_ptr);
+        CHECK(ratio < int_ptr);
+        CHECK(real_ptr > ratio);
+        CHECK(int_ptr > ratio);
+      }
+    }
+
+    TEST_CASE("Edge Case Interactions")
+    {
+      SUBCASE("Ratio Divided by Zero")
+      {
+        auto result = obj::ratio_data(1, 2) / 0.0L;
+        CHECK(std::isinf(result));
+        CHECK(result > 0);
+        CHECK(result == std::numeric_limits<double>::infinity());
+
+        auto neg_result = expect_object<obj::ratio>(-1LL * obj::ratio_data(1, 2))->data / 0.0L;
+        CHECK(std::isinf(neg_result));
+        CHECK(neg_result < 0);
+        CHECK(neg_result == -std::numeric_limits<double>::infinity());
+
+        CHECK_THROWS_AS((obj::ratio_data(1, 2) / 0LL), std::invalid_argument);
+        CHECK_THROWS_AS((obj::ratio_data(1, 2) / obj::ratio_data(0, 1)), std::invalid_argument);
+      }
+
+      SUBCASE("Ratio Multiplied by Negative Integer")
+      {
+        auto result = expect_object<obj::ratio>(obj::ratio_data(2, 3) * -4LL);
+        CHECK(result->data.numerator == -8);
+        CHECK(result->data.denominator == 3);
+      }
+    }
+  }
+
+  TEST_SUITE("ratio")
+  {
+    obj::ratio_ptr make_ptr(native_integer num, native_integer denom)
+    {
+      auto r(make_box<obj::ratio>());
+      r->data.numerator = num;
+      r->data.denominator = denom;
```

Comment:
Please use a ctor to initialize `obj::ratio` with numerator/denominator instead. Then you don't need this whole `make_ptr` fn, since you can just use `make_box`.

Reply:
updated.